src/crush/mapper.c

```diff
@@ -293,10 +293,17 @@ static __u64 crush_ln(unsigned int xin)
 /*
  * straw2
  *
- * for reference, see:
+ * Suppose we have two osds: osd.0 and osd.1, with weight 8 and 4 respectively, It means:
+ *   a). For osd.0, the time interval between each io request apply to exponential distribution 
+ *       with lamba equals 8
+ *   b). For osd.1, the time interval between each io request apply to exponential distribution 
+ *       with lamba equals 4
+ *   c). If we apply to each osd's exponential random variable, then the total pgs on each osd
+ *       is proportional to its weight.
  *
- * http://en.wikipedia.org/wiki/Exponential_distribution#Distribution_of_the_minimum_of_exponential_random_variables
+ * for reference, see:
```

Comment:
maybe worth keeping an additional note about the properties of the min?

Reply:
I think it has little to do with `the properties of the min`,  each osd draws straw by following its own pseudo-random exponential distribution sequence. Osds share nothing between each other.

Reply:
the whole reason we use this distribution is that if you pick the min (max, if negative) after dividing by the weight then the resulting distribution *of placements* is proportional to the weight.  that isn't the case if you use a uniform distribution for the per-osd random raw... it's the whole reason we bother with ln() etc.

Reply:
the formula coincidence with exponential distribution~~
see:  http://preshing.com/20111007/how-to-generate-random-timings-for-a-poisson-process/
and my test:
https://gist.github.com/yaozongyou/53e3d7722c1f1d6dcddb4675206ee65a.

the process can be considered as each osd draws straws by following its own pseudo-random exponential distribution sequence. and at last, we compare each osd' straw, and the longest straw wins. And by the exponential distribution math principle, the min random means happening faster, so the min exponential distribution random variable, the longer straw.